src/main/java/edu/njit/jerse/automation/ProjectRootFinder.java

```diff
@@ -0,0 +1,90 @@
+package edu.njit.jerse.automation;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The {@code ProjectRootFinder} class is responsible for scanning a given directory
+ * and its subdirectories to identify Java project roots. It primarily looks for directories
```

Comment:
"primarily" implies that it is not unique.  Please discuss the other mechanisms, too.